### PR TITLE
Add support for QuickTake 100/150

### PR DESCRIPTION
**Pull request by colinleroy**
- Summary
- Get/Set configuration
- Get preview
- Get PPM files
- Get raw files (QTK format)

Serial protocol reference: https://www.colino.net/wordpress/en/archives/2023/10/29/the-apple-quicktake-100-serial-communication-protocol/ QTKT / QTKN decoding algorithms from dcraw (GPL licensed).

Running on gtkam:
qtkam-qt100

Example photo (after conversion from ppm):
Image_03

### CODE REVIEW

File: camlibs/quicktake1x0/qtkt-decoder.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/* qtkt-decoder.c
+ *
+   Copyright 1997-2018 by Dave Coffin, dcoffin a cybercom o net
+ * Copyright 2023, Colin Leroy-Mira <colin@colino.net>
+ *
+ * QTKT decoder heavily inspired from dcraw.c.
+ *
+ * This file is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
```
Comment (hfiguiere) [inline camlibs/quicktake1x0/qtkt-decoder.c:9]: libgphoto2 is LGPL. That's a problem.

Comment (colinleroy) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: I was wondering about this too, but thought it should be okay given the number of files mentioning GPL in camlibs:

```
grep -lr "GNU General Public License" camlibs/
```

If it isn't, I can drop ppm conversion and only implement .qtk export.

Comment (hfiguiere) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: also note that dcraw is not GPL, expect the "restricted" section which relate only to foveon.
```
No license is required to download and use dcraw.c.  However,
   to lawfully redistribute dcraw, you must either (a) offer, at
   no extra charge, full source code* for all executable files
   containing RESTRICTED functions, (b) distribute this code under
   the GPL Version 2 or later, (c) remove all RESTRICTED functions,
   re-implement them, or copy them from an earlier, unrestricted
   Revision of dcraw.c, or (d) purchase a license from the author.
```

Comment (colinleroy) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: Yes, I've read that paragraph a few times and I *do think* it means I can relicense those two files as LGPL if you'd prefer. However I'm not a lawyer, so I preferred to play it safe after seeing quite a lot of GPL-licensed files in camlibs. 
If you prefer me to relicense as LGPL and also think I'm in the clear doing it, I'll gladly do it!

Comment (hfiguiere) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: not sure about these other camlibs files :-/ it's either a mistake or the pandora box has been open.

Comment (hfiguiere) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: > Yes, I've read that paragraph a few times and I _do think_ it means I can relicense those two files as LGPL if you'd prefer. However I'm not a lawyer, so I preferred to play it safe after seeing quite a lot of GPL-licensed files in camlibs. If you prefer me to relicense as LGPL and also think I'm in the clear doing it, I'll gladly do it!

This is "(c) remove all RESTRICTED functions, re-implement them, or copy them from an earlier, unrestricted Revision of dcraw.c": the restricted code is not part of this. The restricted code starts at line 3046 and end at 3748 in the original dcraw 1.478.

Comment (colinleroy) [reply camlibs/quicktake1x0/qtkt-decoder.c:9]: Yes, and it's an "either".
I'll update the headers tomorrow morning

**Comment by msmeissn** [comment]
is this ready to merge? sorry, did not yet have time to take a longer look

**Comment by colinleroy** [comment]
Hi @msmeissn! From my point of view, yes. Maybe your team will have remarks though?

**Comment by msmeissn** [comment]
thanks, i merged it after the feedback!

**Pull request merged**